packages/core/test/Synchronizer.ts

```diff
@@ -0,0 +1,198 @@
+// @ts-ignore
+import { ethers } from 'hardhat'
+import { BigNumberish, BigNumber } from 'ethers'
+import { expect } from 'chai'
+import { formatProofForSnarkjsVerification } from '@unirep/circuits'
+import * as config from '@unirep/circuits'
+import { genEpochKey, schema } from '@unirep/core'
+import { EpochKeyProof } from '@unirep/contracts'
+import { deployUnirep } from '@unirep/contracts/deploy'
+import { ZkIdentity, genRandomSalt, hashLeftRight } from '@unirep/crypto'
+import { defaultProver } from '@unirep/circuits/provers/defaultProver'
+import { EPOCH_LENGTH } from '@unirep/circuits'
+
+import { maxReputationBudget } from '../config/socialMedia'
+import { deployUnirepSocial, UnirepSocial } from '../src/utils'
+import { genUserState, submitUSTProofs } from './utils'
+
+import { UnirepSocialSynchronizer } from '../src/synchronizer'
+
+import { SQLiteConnector } from 'anondb/node'
+
+let synchronizer: UnirepSocialSynchronizer
+const attestingFee = BigNumber.from(1)
+
+describe('Synchronzier processes events', function () {
+    this.timeout(0)
+
+    let unirepContract
+    let unirepSocialContract: UnirepSocial
+
+    before(async () => {
+        const accounts = await ethers.getSigners()
+        unirepContract = await deployUnirep(accounts[0], {
+            attestingFee,
+        })
+        unirepSocialContract = await deployUnirepSocial(
+            accounts[0],
+            unirepContract.address,
+            {
+                airdropReputation: 30,
+            }
+        )
+        const db = await SQLiteConnector.create(schema, ':memory:')
+        synchronizer = new UnirepSocialSynchronizer(
+            db,
+            defaultProver,
+            unirepContract,
+            unirepSocialContract
+        )
+        // now create an attester
+        await unirepContract
+            .connect(accounts[1])
+            .attesterSignUp()
+            .then((t) => t.wait())
+        await synchronizer.start()
+    })
+
+    afterEach(async () => {
+        await synchronizer.waitForSync()
+        const state = await genUserState(
+            synchronizer.unirepContract.provider,
+            synchronizer.unirepContract.address,
+            new ZkIdentity()
+        )
+        await state.stop()
+    })
```

Comment:
should have a `await compareDB((state as any)._db, (synchronizer as any)._db)` here
or we can ignore the `genUserState`

Reply:
not sure I understand why we should ignore `genUserState` here

Reply:
we want to compare the state with `waitForSync()` is the same as `genUserState()` from the start block.
(To make sure the `waitForSync()` works well)